File: task2-peer/src/main/scala/marmik/sbc/task2/peer/SuperPeer.scala
package marmik.sbc.task2.peer

trait SuperPeer {
  
  /** 
   * @param url: Url of super pper
   * @param selfName: Name of this peer
   * @param url: Url of this peer 
   */
  def login(url:String, selfName:String, selfUrl:String);

  /**
   Registers a listener to get notification about new posts or edits.
   */
  def register(l:Listener);

  /** Returns list of available peers
   */
  def getPeers():List[Peer];
  
  def logout();  
  
}

File: task2-peer/src/main/scala/marmik/sbc/task2/rmi/SuperPeer.java
package marmik.sbc.task2.rmi;

import java.util.List;

public interface SuperPeer extends java.rmi.Remote {
	public List<Topic> getTopcis() throws java.rmi.RemoteException;
	public void newTopic(String url, String name) throws java.rmi.RemoteException;
}
